docker.py

```diff
@@ -40,6 +45,7 @@ def run(self):
                 err_path = os.path.join(log_path, "err.log")
                 with open(out_path, "a") as out_fd:
                     with open(err_path, "a") as err_fd:
+                        kwargs.update(stdout=out_fd, stderr=err_fd)
```

Comment:
why do we change kwargs after it is used?

Reply:
Ah :see_no_evil: you spotted how ugly this is. I didn't feel like making it nice. But now I feel exposed and want to.
For explanation `kwargs` is a dict that bound to the whole scope of the `_subprocess_run` function and `_process_run` uses those kwargs. So I'm modifying the `kwargs` dict in place to change what `_process_run` does. It's as ugly and hard to follow as using global variables :nauseated_face: